### ISSUE TITLE
Extend no-empty-function to handle constructors with parameter properties

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#426]
Spawning off of #48 for clarity.

# What code were you trying to parse?
```TS
export default class Foo {
  constructor(private name: string) {}
}
```

# What happened?
```
  2:36  error  Unexpected empty constructor  no-empty-function
```

**Comment by Harris-Miller** [comment]
+1

**Comment by eyedean** [comment]
I was hoping for something like `@typescript-eslint/no-empty-function` to exist, just like what [fixed `no-useless-constructor`](https://github.com/typescript-eslint/typescript-eslint/issues/48#issuecomment-460110630).  I believe that's what this issue is for! :)

Well, until that comes to existence, I am going to go with the following in my `.eslintrc.js`:

```json
   "no-empty-function": ["error", { "allow": ["constructors"] }],
```

PS. Sorry for treating here like Stackoverflow.com.  I'll come back and remove this temporary solution once this issue is resolved!

**Comment by scottohara** [comment]
This hypothetical `@typescript-eslint/no-empty-function` rule extension would presumably allow for:

* `if node.kind === "constructor"`, and
* `node.body.body.length == 0`, and
* `node.value.params.some(param => param.type === AST_NODE_TYPES.TSParameterProperty)`

(in other words: IF the function is a constructor AND it has an empty body AND it has one or more parameter properties...don't flag it as an error).

If I can find some spare time over the next few days, I'll try to get a PR started.
(no promises, though)

**Comment by scottohara** [comment]
PR is now up (#626)
